tests/kernel/fatal/exception/src/main.c

```diff
@@ -138,6 +170,19 @@ void entry_arbitrary_reason(void *p1, void *p2, void *p3)
 	irq_unlock(key);
 }
 
+void entry_arbitrary_reason_negative(void *p1, void *p2, void *p3)
+{
+	unsigned int key;
+
+	expected_reason = -2;
+
+	key = irq_lock();
```

Comment:
What was the need for locking IRQs here?

Reply:
I copy this lock IRQ from other testcase in this file.
I think the author write this code here is because expected_reason is a global variable,  use irq_lock() here will explicitly explain that the variable would not be changed in the isr context.